src/charm.py

```diff
@@ -120,6 +124,11 @@ def __init__(self, *args):
         self.framework.observe(self.workload_tracing.on.endpoint_removed, self._reconcile)
         self.framework.observe(self.on.collect_unit_status, self.on_collect_status)
 
+        self.framework.observe(self.on["istio-ingress-config"].relation_joined, self._reconcile)
+        self.framework.observe(self.on["istio-ingress-config"].relation_changed, self._reconcile)
+        self.framework.observe(self.on["istio-ingress-config"].relation_broken, self._reconcile)
+        self.framework.observe(self.on["istio-ingress-config"].relation_departed, self._reconcile)
+
```

Comment:
This side of the relation just responds to requests, right?  Eg: ingress sends us data, and based on that we configure some things and send a response back?  If true, all we need here is relation_changed.

On the other hand, I can see an argument that we should just reconcile on everything regardless.  If that's what you're going for, I'm like a tiny bit against it but lgtm 

```suggestion
        self.framework.observe(self.on["istio-ingress-config"].relation_changed, self._reconcile)

```

Reply:
I think we also need `relation_broken` to reconcile on relation removal?

Reply:
Yeah good point.  My memory was that we get relation_changed (with the data removed) before the departed/broken, but looking at the docs I think I'm wrong.  So then we need to track `relation_changed` and `relation_departed` (eg: app is gone), but not the others